src/System.Net.Sockets/src/System/Net/Sockets/SendPacketsElement.cs

```diff
@@ -39,7 +43,40 @@ public SendPacketsElement(string filepath, int offset, int count, bool endOfPack
                 throw new ArgumentOutOfRangeException(nameof(count));
             }
 
-            Initialize(filepath, null, offset, count, SendPacketsElementFlags.File, endOfPacket);
+            Initialize(filepath, null, null, offset, count, endOfPacket);
+        }
+
+        // Constructors for fileStream elements.
+        public SendPacketsElement(FileStream fileStream) :
+            this(fileStream, 0L, 0, false)
+        { }
+
+        public SendPacketsElement(FileStream fileStream, long offset, int count) :
+            this(fileStream, offset, count, false)
+        { }
+
+        public SendPacketsElement(FileStream fileStream, long offset, int count, bool endOfPacket)
+        {
+            // We will validate if the fileStream exists on send.
+            if (fileStream == null)
+            {
+                throw new ArgumentNullException(nameof(fileStream));
+            }
+            if (!fileStream.IsAsync)
+            {
+                throw new ArgumentException(nameof(fileStream), SR.GetResourceString("net_sockets_sendpackelement_FileStreamMustBeAsync"));
```

Comment:
`SR.GetResourceString("net_sockets_sendpackelement_FileStreamMustBeAsync")` should instead be `SR.net_sockets_sendpackelement_FileStreamMustBeAsync`